### PR TITLE
Preparing for Elephant release 0.4.0 

### PR DESCRIPTION
**Pull request by alperyeg**
For the upcoming neo 0.5.0 release I prepared in parallel the next release for Elephant and adjusted the versions, docs etc. Let me know if something is missing.

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -15,15 +15,13 @@ The following packages are required to use Elephant:
     * numpy_ >= 1.8.2
     * scipy_ >= 0.14.0
     * quantities_ >= 0.10.1
-    * neo_ == 0.4.0
-
-.. note:: at the time of writing Neo 0.4.0 has not been released. You should therefore use `this snapshot`_.
+    * neo_ == 0.5.0
```
Comment (apdavison) [inline doc/install.rst]: I suggest making this `>=`, so when we release Neo 0.5.1, etc., we don't need to change this.

File: doc/release_notes.rst [hunk 2]
```diff
@@ -2,6 +2,32 @@
...
+=============
+* `spike_train_generation` module:
+    * peak detection: **peak_detection()**
+* Modules for Current Source Density: `current_source_density_src`
+    * Module Current Source Density: `KCSD.py`
+    * Module for Inverse Current Source Density: `icsd.py`
+
+API changes
+===========
+* Interoperability between Neo 0.5.0beta1 and Elephant
```
Comment (apdavison) [inline doc/release_notes.rst]: remove "beta1"

File: doc/requirements.txt [hunk 3]
```diff
@@ -1,7 +1,6 @@
 # Requirements for building documentation
 numpy>=1.8.2
 quantities>=0.10.1
--e git+https://github.com/NeuralEnsemble/python-neo.git@snapshot-20150821#egg=neo-snapshot-20150821
-# note that we use a mock scipy
+neo>=0.4.1
```
Comment (apdavison) [inline doc/requirements.txt]: >=0.5.0

**Comment by alperyeg** [comment]
@apdavison thanks for the quick reply and corrections!

**Pull request merged**